### PR TITLE
Adding flux-cli to zarf.yaml

### LINKED ISSUE #131: Add Flux CLI
I've found myself downloading the Flux CLI a few different times now to troubleshoot issues. It might be nice to just add it to the package as an optional component.

### PR DESCRIPTION
**Pull request by matt-strong**
Fixes #131

### CODE REVIEW

File: zarf.yaml [hunk 1]
```diff
@@ -179,6 +179,15 @@ components:
       - kiwigrid/k8s-sidecar:1.14.2
       # - registry.opensource.zalan.do/acid/spilo-13:2.1-p1 -- Already present in softwarefactoryaddons-deps component
 
+  - name: flux-cli
+    required: false
+    description: Flux is a tool for keeping Kubernetes clusters in sync with sources of configuration (like Git repositories), and automating updates to configuration when there is new code to deploy.
+    files:
+      - source: https://github.com/fluxcd/flux2/releases/download/v0.31.1/flux_0.31.1_linux_amd64.tar.gz
+        shasum: 640d2f9b403f446fbb7ee7b99bf58341aa1a98d64688b9f47e533b78dd212166
+        target: flux.tar.gz
```
[the remark this answers is not in the document]
Comment (matt-strong) [reply zarf.yaml:188]: Yeah, that's all I'm seeing here: https://github.com/fluxcd/flux2/releases

[the remark this answers is not in the document]
Comment (RothAndrew) [reply zarf.yaml:188]: Works for me

File: zarf.yaml [hunk 2]
```diff
@@ -179,6 +179,18 @@ components:
...
+    required: false
+    scripts:
+      after:
+        - "./zarf tools archiver decompress flux.tar.gz /usr/sbin/flux"
+        - "chmod +x /usr/sbin/flux"
+    files:
+      - source: https://github.com/fluxcd/flux2/releases/download/v0.31.1/flux_0.31.1_linux_amd64.tar.gz
+        shasum: 640d2f9b403f446fbb7ee7b99bf58341aa1a98d64688b9f47e533b78dd212166
+        target: flux.tar.gz
+        executable: true
```
Comment (RothAndrew) [inline zarf.yaml:192]: This can be false since we don't need the .tar.gz file to be executable, just the binary that gets extracted

**Comment by RothAndrew** [comment]
Let's also add a line to the E2E test that tests whether `flux --help` or something returns a zero exit code

File: zarf.yaml [hunk 3]
```diff
@@ -179,6 +179,18 @@ components:
       - kiwigrid/k8s-sidecar:1.14.2
       # - registry.opensource.zalan.do/acid/spilo-13:2.1-p1 -- Already present in softwarefactoryaddons-deps component
 
+  - name: flux-cli
+    required: false
+    scripts:
+      after:
+        - "./zarf tools archiver decompress flux.tar.gz /usr/sbin/flux"
```
Comment (RothAndrew) [inline zarf.yaml:186]: We can't depend on the Zarf binary being in the same folder where this is being run. Using `tar` here instead I think is valid, as it is an extremely common tool on all *nix operating systems.

I think there might also be some magic here that we could utilize. @jeff-mccoy if we just put `zarf` here (instead of `./zarf` will Zarf automatically know where that binary is? I'm remembering back to a while ago that makes me believe Zarf may do some magical pathfinding stuff here

File: zarf.yaml [hunk 4]
```diff
@@ -179,6 +179,18 @@ components:
...
+        - "chmod +x /usr/sbin/flux"
```
[the remark this answers is not in the document]
Comment (matt-strong) [reply zarf.yaml:187]: works for me

File: test/e2e/e2e_basic_smoke_test.go [hunk 5]
```diff
@@ -75,3 +75,19 @@ func TestAllServicesRunning(t *testing.T) {
...
+	ctx, cancel := context.WithCancel(context.Background())
+	go utils.HoldYourDamnHorses(ctx, t, 10*time.Second)
+	defer cancel()
+	platform := types.NewTestPlatform(t)
+	defer platform.Teardown()
+	utils.SetupTestPlatform(t, platform)
+	// The repo has now been downloaded to /root/app and the software factory package deployment has been initiated.
+	teststructure.RunTestStage(platform.T, "TEST", func() {
+		output, err := platform.RunSSHCommandAsSudo("flux --help")
+		require.NoError(t, err, output)
```
Comment (RothAndrew) [inline test/e2e/e2e_basic_smoke_test.go:91]: Let's just add these 2 lines to the existing test. Doing it this way will cause 2 instances to be spun up which will significantly increase cloud costs and IO pressure on Registry1

**Review by RothAndrew — APPROVED** [review]
Looks great! Thanks!
